prov/shm/src/smr_progress.c

```diff
@@ -1273,5 +1273,6 @@ void smr_ep_progress(struct util_ep *util_ep)
 
 	/* always drive forward the ipc list since the completion is
 	 * independent of any action by the provider */
-	smr_progress_ipc_list(ep);
+	if (ep->smr_async_device_copy)
+		smr_progress_ipc_list(ep);
```

Comment:
Rather than adding a check here, we can setup a function pointer that either calls a no-op or smr_progress_ipc_list().  No-op functions are typically faster than adding an if statement and doesn't add a check when we do need to progress the ipc list.